test/dynamo/test_optimizers.py

```diff
@@ -29,6 +29,9 @@ def get_optimizer_step(opt, closure=None):
     # see the [Note on graph break] in optimizer.py
     # This ignores the outer _use_grad_if_differentiable wrapper, which is fine for now
     # as dynamo does not support differentiable optimizers anyway
+    if not hasattr(opt.step, "__wrapped__"):
+        return lambda: opt.step
```

Comment:
```suggestion
        step_fn = opt.step.__func__
```

and then have line 35 be in an else.

This feels more correct (would handle closure, I believe).

Reply:
OKAY. I have a better fix that won't require this hack at all. 

In `make_test`, instead of the following,
```
def make_test(optim_cls, closure=None, **kwargs):
    opt = optim_cls(model.parameters(), **kwargs)

    def test_fn(self):
        nonlocal opt
```
You should change this to
```
def make_test(optim_cls, closure=None, **kwargs):
    def test_fn(self):
        opt = optim_cls(model.parameters(), **kwargs)
```

Why? The TorchPatcher in torch/_dynamo/eval_frame.py overwrites optimizers' attributes and thus reusing the opt in make_test conflates the opt object for the new AdamW.

Yes this feels shady, though it is out of scope for this PR to look into that further.

Reply:
Sounds great, I'll commit this in a moment!

Reply:
Just pushed changes, is there anything to be done in test_compiled_optimizers? Don't think opt is reused there

Reply:
test_compiled_optimizers already does something similar to this haha

Reply:
Looks like there are some failures :(. Should I run the (updated) hacks against the CI again?